src/Microsoft.DotNet.Interactive/Connection/ProxyKernel.cs

```diff
@@ -64,16 +64,56 @@ public sealed class ProxyKernel : Kernel
         RegisterForDisposal(subscription);
     }
 
-    private void UpdateKernelInfoFromEvent(KernelInfoProduced kernelInfoProduced)
+    public ProxyKernel(
+        string name,
+        IKernelCommandAndEventSender sender,
+        IKernelCommandAndEventReceiver receiver,
+        Uri remoteUri,
+        string displayName,
+        bool isComposite,
+        string languageName,
+        string languageVersion,
```

Comment:
Do we need all of these properties in the constructor? I would think we only need the ones where the remote and the proxy might be different from one another. Or we could explore the earlier proposal which is just to make `Kernel.Name` mutable.

Reply:
I would support the mutable approach and just make sure the composite kernel collection is able to see that and react.

Reply:
Pat of those parameters are already mutable on the kernel info

Reply:
Yeah, generally constructor parameters for mutable values don't add value.

Reply:
Yes, that totally makes sense. The main reason to create this constructor was so that we could also set `_requiresRequestKernelInfoOnFirstCommand` to false in the case where a `ProxyKernel` is being created based on results of `RequestKernelInfo` command. I previously had a constructor that accepted the corresponding `KernelInfo` directly - however based on Jon's feedback when we discussed this offline, we decided to switch to passing the underlying arguments instead.

I'll go ahead and remove the new constructor. However, I think it is still valuable to set `_requestKernelInfoOnFirstCommand` to `false` in the case where we already have the `KernelInfo`.

I can introduce an optional flag on the existing constructor for this OR we could remove this flag entirely and make it implicit (e.g. if `SupportedCommands` / `SupportedDirectives` is empty then trigger `RequestKernelInfo` implicitly before the first command inside `EnsureKernelInfoIsLoaded`).

Thoughts?

Reply:
I think this should be explicit. We shouldn't do anything implicitly that introduces a network call, and there are good reasons (e.g. performance) to avoid waiting on the network at startup.

> I previously had a constructor that accepted the corresponding KernelInfo directly - however based on Jon's feedback when we discussed this offline, we decided to switch to passing the underlying arguments instead.

I think the only argument that really matters here is the kernel name, but that's only because we made the property immutable. Making it mutable is easier to understand I think.

Reply:
Thanks - I have reverted the constructor.

I haven't changed mutability of `Kernel.Name` (and `KernelInfo.LocalName`) yet since this does not appear necessary yet (since `StdIoKernelConnector` now allows to override name for any kernel when it is created).

One minor wrinkle with making them mutable is that we need to ensure that `KernelInfo.LocalName` and `Kernel.Name` should match. And this would also impact `Kernel.DisplayName` (which is currently copied over from the remote side) - so we may also need a way to update the `DisplayName` to match the new `Name` (if the `Name` appears inside `DisplayName`).

I don't think this is blocking for the current PR though - we can always improve how we handle collisions more in the future. But let me know if you feel strongly that we should make the names mutable in the current PR.